### ISSUE TITLE
Weave 2.0 crashes when launched with password and ip (with TLS enabled in Docker)

### ISSUE THREAD

**Issue by stuartz-VernonCo** [weaveworks/weave#3034]
Confirmed splitting commands as a work around from Bryan Boreham works:

 ```
weave launch --ipalloc-init observer --ipalloc-range 10.36.0.0/16 --password=$WEAVE_PASSWORD 
 weave connect $PREVIOUS_IP
```

++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    weave crashes.  weave is looking for directory /home/weave/tls  which I have never had on my other instances

    #>whoami

    ubuntu

    #>sudo curl -L git.io/weave -o /usr/local/bin/weave
    #>sudo chmod +x /usr/local/bin/weave

    #>weave launch --ipalloc-init observer --ipalloc-range 10.36.0.0/16 \
        --password=$WEAVE_PASSWORD $PREVIOUS_IP

    #>weave version

    weave script 2.0.0
    weave 2.0.0

    #>docker version

    Client:
     Version:      17.05.0-ce
     API version:  1.29
     Go version:   go1.7.5
     Git commit:   89658be
     Built:        Thu May  4 22:10:54 2017
     OS/Arch:      linux/amd64

    Server:
     Version:      17.05.0-ce
     API version:  1.29 (minimum version 1.12)
     Go version:   go1.7.5
     Git commit:   89658be
     Built:        Thu May  4 22:10:54 2017
     OS/Arch:      linux/amd64
     Experimental: false

    #>docker logs weave  (keeps running the following ever 1 minute)

    INFO: 2017/06/23 17:06:10.026779 Command line options: map[dns-effective-listen-address:172.17.0.1 ipalloc-init:observer host-root:/host nickname:node-dev3 plugin:true status-addr:127.0.0.1:6782 datapath:datapath H:[unix:///var/run/weave/weave.sock] http-addr:127.0.0.1:6784 ipalloc-range:10.36.0.0/16 proxy:true dns-listen-address:172.17.0.1:53 docker-bridge:docker0 port:6783 resolv-conf:/var/run/weave/etc/resolv.conf weave-bridge:weave]
    INFO: 2017/06/23 17:06:10.026863 weave  2.0.0
    INFO: 2017/06/23 17:06:10.028386 Docker API on unix:///var/run/docker.sock: &[GitCommit=89658be GoVersion=go1.7.5 Os=linux Arch=amd64 BuildTime=2017-05-04T22:10:54.638119411+00:00 Version=17.05.0-ce ApiVersion=1.29 MinAPIVersion=1.12 KernelVersion=4.4.0-57-generic]
    INFO: 2017/06/23 17:06:10.028519 Using docker bridge IP for DNS: 172.17.0.1
    INFO: 2017/06/23 17:06:10.031959 proxy listening on unix:///var/run/weave/weave.sock
    INFO: 2017/06/23 17:06:10.033192 Bridge type is bridged_fastdp
    INFO: 2017/06/23 17:06:10.033210 Communication between peers via untrusted networks is encrypted.
    INFO: 2017/06/23 17:06:10.076214 Our name is ce:5d:e1:45:98:99(node-dev3)
    INFO: 2017/06/23 17:06:10.076275 Launch detected - using supplied peer list: [173.215.44.190 --tlsverify --tlscacert /home/weave/tls/ca.pem --tlscert /home/weave/tls/cert.pem --tlskey /home/weave/tls/key.pem]
    INFO: 2017/06/23 17:06:10.077529 Docker API on unix:///var/run/docker.sock: &[Version=17.05.0-ce GitCommit=89658be BuildTime=2017-05-04T22:10:54.638119411+00:00 Arch=amd64 KernelVersion=4.4.0-57-generic ApiVersion=1.29 MinAPIVersion=1.12 GoVersion=go1.7.5 Os=linux]
    INFO: 2017/06/23 17:06:10.078117 Checking for pre-existing addresses on weave bridge
    INFO: 2017/06/23 17:06:10.107199 [allocator ce:5d:e1:45:98:99] Initialising with persisted data
    INFO: 2017/06/23 17:06:10.120162 Listening for DNS queries on 172.17.0.1
    INFO: 2017/06/23 17:06:10.139729 Sniffing traffic on datapath (via ODP)
    FATA: 2017/06/23 17:06:10.141829 lookup --tlsverify: no such host
    lookup --tlscacert: no such host
    lookup /home/weave/tls/ca.pem: no such host
    lookup --tlscert: no such host
    lookup /home/weave/tls/cert.pem: no such host
    lookup --tlskey: no such host
    lookup /home/weave/tls/key.pem: no such host

**Comment by bboreham** [comment]
For clarity this is a mistake in how the command-line arguments for weaver and weaveproxy were merged.  `/home/weave/tls` is inside the container's filesystem.

**Comment by marccarre** [comment]
Note that it seems to only happen with TLS enabled in Docker.

**Comment by marccarre** [comment]
Issue successfully reproduced after following:

- instructions on https://docs.docker.com/engine/security/https/ to enable TLS in Docker
- replacing:
```
# cat /etc/systemd/system/docker.service.d/docker_over_tcp.conf
[Service]
ExecStart=
ExecStart=/usr/bin/docker daemon -H fd:// -H unix:///var/run/alt-docker.sock -H tcp://0.0.0.0:2375 -s overlay
```
  with:
```
# cat /etc/systemd/system/docker.service.d/docker_over_tcp.conf
[Service]
ExecStart=
ExecStart=/usr/bin/docker daemon -H fd:// -H unix:///var/run/alt-docker.sock -H tcp://0.0.0.0:2375 -s overlay --tlsverify --tlscacert=/etc/docker/ca.pem --tlscert=/etc/docker/server-cert.pem --tlskey=/etc/docker/server-key.pem
```
- and running:
```
# systemctl daemon-reload
# systemctl restart docker
$ export WEAVE_PASSWORD="s3cr3t"
$ export PREVIOUS_IP="173.215.44.190"
$ weave launch --ipalloc-init observer --ipalloc-range 10.36.0.0/16 \
    --password=$WEAVE_PASSWORD $PREVIOUS_IP
```

**Comment by marccarre** [comment]
Fixed by #3039
